VC14 Cleanup

GregoryLundberg opened the pull request:
These commits should get us to 0 errors / 0 warnings for both Release and Debug builds for Visual Studio 2015.

In general there were three classes of error:

    * A local name hiding another. Renamed the local name

    * The poorly-written logging macro causing a duplicated name to hide an earlier name. Removed the duplicate macro. This eliminated some inner log scopes. A better fix would be to completely replace the lame macro with a better one; but that is too much work for today.

    * Bugs in spirit_po and certain Boost libraries. These were names-hides-name errors. Some could have been fixed by renaming some Wesnoth global variables; some are true bugs in the libraries. Since these are libraries, they messages were suppressed rather than correcting the library.

Vultraz commented:
Don't merge this yet, I'd like to check why I declared two data/item vars in mp_staging.

Vultraz commented:
See 82a8797

Vultraz commented:
Wait, 87227575831f. Sorry >_>

CelticMinstrel commented on src/gettext_boost.cpp:
```diff
@@ -153,18 +153,18 @@ namespace
 		}
 
-		const char* get(int domain_id, const char* ctx, const char* id) const override
+		const char* get(int domain_id, const char* ctx, const char* id_param) const override
```
What about `msg_id`? Seems more descriptive.

GregoryLundberg replied:
OK

CelticMinstrel commented on src/gui/widgets/window.cpp:
```diff
@@ -1243,8 +1243,8 @@ void swap_grid(grid* g,
 	}
-	if(grid* g = dynamic_cast<grid*>(parent_grid->parent())) {
-		widget = g->swap_child(id, widget, false);
+	if(grid* g_in_if = dynamic_cast<grid*>(parent_grid->parent())) {
```
This is a really terrible name. I think I'd prefer renaming the parameter instead in this case. Not entirely sure what's a good name though. Maybe the doxy-comment in the header can give a good idea.

CelticMinstrel commented on src/gettext_boost.cpp:
```diff
@@ -11,6 +11,9 @@
    See the COPYING file for more details.
 */
+#if defined(_MSC_VER)
+#pragma warning(disable: 4714)
```
Wait, why did you move this up here instead of just adding another warning to the other pragma?

GregoryLundberg replied:
It was originally there. I moved it to make it apparent it's fully file-scoped. Note no pop/push either.
The warning occurs during link-time code generation.

CelticMinstrel replied:
Pretty sure it was _supposed_ to be just for `spirit_po` though, which is why it was there; so the mistake is probably forgetting the push/pop.

GregoryLundberg commented on src/gettext_boost.cpp:
```diff
@@ -30,7 +33,7 @@
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
 #elif defined(_MSC_VER)
 #pragma warning(push)
-#pragma warning(disable: 4714)
```
4714 was originally down here
4459 was needed so I added it
then I noticed 4714 was not working here, so I moved it

GregoryLundberg commented:
1) dropped commit per @Vultraz changing
2) renamed variables to better names
3) put logging scopes back in, hand-expanded lame macro to avoid errors

This should address all comments.

CelticMinstrel commented:
Well, I don't like hand-expanding the macro, but I guess it's better than removing the log scopes altogether.

GregoryLundberg commented:
Actually, I don't like anything about the macro and want to re-write it but with everything going on here, I am trying to do small things and not think too hard.

Fixing the macros should be a separate subject PR. So consider the hand-expansion as hand-waving to point out it needs doing.
